tests/tests_pytorch/utilities/test_data.py

```diff
@@ -173,6 +174,30 @@ def __init__(self, randomize, *args, **kwargs):
     assert isinstance(new_dataloader, GoodImpl)
 
 
+def test_replace_init_method_multiple_loaders_without_init():
+    """In case of a class, that inherits from a class that we are patching, but doesn't define its own `__init__`
+    method (the one we are wrapping), it can happen, that `hasattr(cls, "_old_init")` is True because of parent
+    class, but it is impossible to delete, because that method is owned by parent class. Furthermore, the error
+    occured only sometimes because it depends on the order in which we are iterating over a set of classes we are
+    patching.
+
+    This test simulates the behavior by generating sufficient number of dummy classes, which do not define `__init__`
+    and are children of `DataLoader`. We are testing that a) context manager `_replace_init_method` exits cleanly, and
+    b) the mechanism checking for presence of `_old_init` works as expected.
+    """
+    classes = [DataLoader]
+    for i in range(100):
+        classes.append(type(f"DataLoader_{i}", (random.choice(classes),), {}))
+
+    with _replace_init_method(DataLoader, "dataset"):
+        for cls in classes:
+            if cls == DataLoader:
+                assert "_old_init" in cls.__dict__
```

Comment:
Only one class is `DataLoader` right? So this can be moved outside of the `for` loop

Also class equality is usually checked with `is`

Reply:
Yep, only one is `DataLoader`. Resolved with iterating over all the other